### PR TITLE
Remap the level entity ids with the same values on server and clients

### PR DESCRIPTION
**Pull request by vmednonogov**
## What does this PR do?
Remap the level entity ids with the same values on server and clients

## How was this PR tested?

Verified locally

### CODE REVIEW

**Review by VladimirLagutin — APPROVED** [review]
LGTM

File: Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp [hunk 1]
```diff
@@ -452,6 +452,27 @@ namespace AzFramework
...
+        previouslySpawned.reserve(entities.size());
+
+        for (auto& entity : entities)
+        {
+            idMap.emplace(entity->GetId(), seedEntityId);
+            seedEntityId = AZ::EntityId(static_cast<AZ::u64>(seedEntityId)+1);
+        }
+    }
+#endif
+// Gruber patch begin // VMED
```
Comment (aefimov3) [inline Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp]: ... end

File: Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp [hunk 2]
```diff
@@ -472,6 +493,19 @@ namespace AzFramework
...
+#ifdef CARBONATED
+        constexpr const char* levelSpawnableName = "centralplaza.spawnable"; // FIXME change to evaluated level name or use another way to detect a level spawnable
+        const bool isLevelSpawnable = ::strstr(ticket.m_spawnable.GetHint().c_str(), levelSpawnableName) != nullptr;
+        AZ::EntityId seedEntityId;
+        if (isLevelSpawnable)
+        {
+            seedEntityId = AZ::EntityId(777777700000000); // for "centralplaza" // we have approx 0% probability to collide an existing entity id with the ids group in interval [seedEntityId, seedEntityId+NumOfEntities[
+        }
+#endif
+// Gruber patch begin // VMED
```
Comment (aefimov3) [inline Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp]: ... end

Comment (vmednonogov) [reply Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp]: updated

File: Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp (hunk 1 above)
Comment (vmednonogov) [reply Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp]: updated

**Comment by vmednonogov** [comment]
This branch is kept as a branch for future network support experiments

**Comment by vmednonogov** [comment]
Merged according to Vladimir's request

**Pull request merged**